marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -476,7 +476,11 @@ lintValue env t@(Term SlotIntervalStart pos) = pure (ValueSimp pos false t)
 
 lintValue env t@(Term SlotIntervalEnd pos) = pure (ValueSimp pos false t)
 
-lintValue env t@(Term (UseValue (Term valueId pos2)) pos) = pure (ValueSimp pos false t)
+lintValue env t@(Term (UseValue (Term valueId _)) pos) = do
+  let
+    undefinedLet = if Set.member valueId (view _letBindings env) then mempty else Set.singleton (UndefinedUse (termToRange t pos))
```

Comment:
do you need this `if`? Since you are using a Set can't you just add the singleton whether it is already there or not? Would just make the code a bit simpler...